[DOCS] DOC-341 partition how to configure a new data context with the cli guide

Rachel-Reverie opened the pull request:
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- Partitions each section of the how to configure a new data context guide into individual files in the `components_how_to_configure_a_new_data_context_with_the_cli` folder.
- Partitions each section of the Setup: Index page into individual files in the `components_index` folder.
- Adds content to the "how to configure a new data context with the cli" guide:
  - Renames the guide to something more accurate to its contents.
  - Adds more details on what to expect from the process
  - Adds guidance on how to verify initialization was successful
  - Adds links to  the configuration guides as "Next steps"

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation


Thank you for submitting!

Rachel-Reverie committed:
DOC-341
- Partitions each section of the how to configure a new data context guide into individual files in the `components_how_to_configure_a_new_data_context_with_the_cli` folder.
- Partitions each section of the Setup: Index page into individual files in the `components_index` folder.
- Adds content to the "how to configure a new data context with the cli" guide:
  - Renames the guide to something more accurate to its contents.
  - Adds more details on what to expect from the process
  - Adds guidance on how to verify initialization was successful
  - Adds links to  the configuration guides as "Next steps"

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=5426&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=5426&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/5426/df80fe0b/4b330ca1f18bda1b7f8ef842d80a9dfb4b340027.svg)](https://app.codesee.io/r/reviews?pr=5426&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-341
- Adds partitioned file for generic "congrats" that gets appended to each how-to guide at the end.

Rachel-Reverie committed:
DOC-341
- Updates title of Data Context guide in the Setup: Index files.

rdodev reviewed (approved):
LGTM

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-280/DOC-341/partition_how_to_configure_a_new_data_context_with_the_cli
